app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,566 @@
+      }
+    }
+  }
+
+  public void open(BooksOnDiskListItem.BookOnDisk bookOnDisk) {
+    File file = bookOnDisk.getFile();
+    Intent zimFile = new Intent(this, MainActivity.class);
+    zimFile.setData(Uri.fromFile(file));
+    startActivity(zimFile);
+    finish();
+  }
+
+  public void select(BooksOnDiskListItem.BookOnDisk bookOnDisk) {
+    ArrayList<BooksOnDiskListItem> booksList = new ArrayList<>();
+    for (BooksOnDiskListItem item : booksAdapter.getItems()) {
+      if (item.equals(bookOnDisk)) {
+        if (item.isSelected()) {
+          item.setSelected(false);
+        } else {
+          item.setSelected(true);
+        }
+      }
+      booksList.add(item);
+    }
+    booksAdapter.setItems(booksList);
+  }
+
+  @Override protected void onStart() {
+    super.onStart();
+    bindService();
+  }
+
+  @Override protected void onStop() {
+    super.onStop();
+    unbindService();
+  }
+
+  private void bindService() {
+
+    bindService(serviceIntent, serviceConnection, Context.BIND_AUTO_CREATE);
+  }
+
+  private void unbindService() {
+    if (bound) {
+      hotspotService.registerCallBack(null); // unregister
+      unbindService(serviceConnection);
+      bound = false;
+    }
+  }
+
+  @RequiresApi(api = Build.VERSION_CODES.O)
+  void toggleHotspot() {
+    //Check if location permissions are granted
+    if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION)
+        == PackageManager.PERMISSION_GRANTED) {
+
+      startService(ACTION_IS_HOTSPOT_ENABLED); //If hotspot is already enabled, turn it off
+    } else {
+      //Ask location permission if not granted
+      ActivityCompat.requestPermissions(this,
+          new String[] { Manifest.permission.ACCESS_FINE_LOCATION },
+          MY_PERMISSIONS_ACCESS_FINE_LOCATION);
+    }
+  }
+
+  @Override protected void onResume() {
+    super.onResume();
+    presenter.loadBooks();
+    if (isServerStarted) {
+      ip = getAddress();
+      ip = ip.replaceAll("\n", "");
+      serverTextView.setText(getString(R.string.server_started_message) + " " + ip);
+      startServerButton.setText(getString(R.string.stop_server_label));
+      startServerButton.setBackgroundColor(getResources().getColor(R.color.stopServer));
+    }
+  }
+
+  // This method checks if mobile data is enabled in user's device.
+  static boolean isMobileDataEnabled(Context context) {
+    ConnectivityManager cm =
+        (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+    try {
+      Class cmClass = Class.forName(cm.getClass().getName());
+      Method method = cmClass.getDeclaredMethod("getMobileDataEnabled");
+      method.setAccessible(true);
+      return (Boolean) method.invoke(cm);
+    } catch (Exception e) {
+      Log.e(TAG, e.toString());
+    }
+    return false;
+  }
+
+  //This method sends the user to data usage summary settings activity
+  private void openDataUsageActivity() {
```

Comment:
Does `openMobileDataActivity` seem a better name?

Reply:
I've commented the mobile data check code since long. I'm not sure if reflection is worth the impact for this code. This needs a bit of discussion.
IMO, just giving the user a warning that mobile data is enabled is not worth using reflection for. We can skip that.

Reply:
I think you got me wrong. I'm just asking if we should rename this method to `openMobileDataActivity`.

Reply:
@abdulwd Yes we can rename it to this.

Reply:
Fixed. 1e7c5c9b33ff3dd15343c00025660951b74dfe56